src/Integration.Vsix/SonarLintTagger/CancellableJobRunner.cs

```diff
@@ -26,161 +26,180 @@
 using Microsoft.VisualStudio.Threading;
 using SonarLint.VisualStudio.Integration.Vsix.Resources;
 
-namespace SonarLint.VisualStudio.Integration.Vsix
+namespace SonarLint.VisualStudio.Integration.Vsix.CancellableJob
 {
-    /// <summary>
-    /// Runs a series of operations on a separate background thread.
-    /// Can be cancelled.
-    /// </summary>
-    internal sealed class CancellableJobRunner
+    public interface ICancellableJobRunner
     {
-        // Not overriding Equals etc - this type is only used for progress reporting.
-        // It's not intended to be stored or compared
-        public struct JobRunnerProgress
-        {
-            public JobRunnerProgress(RunnerState currentState, int completed, int totalOperations)
-            {
-                CurrentState = currentState;
-                CompletedOperations = completed;
-                TotalOperations = totalOperations;
-            }
+        /// <summary>
+        /// Runs a series of operations on a separate background thread.
+        /// Can be cancelled.
+        /// </summary>
+        ICancellableJob Start(string jobDescription, IEnumerable<Action> operations, IProgress<JobRunnerProgress> progress, ILogger logger);
```

Comment:
The GitHub diff is confusing and not particularly helpful.
Only the construction code has changed; everything else is the same.
I'll call out the changes in comments.